src/services/inference/module/module_js.ml

```diff
@@ -408,23 +411,45 @@ module Node = struct
 
   let rec node_module ~options ~reader node_modules_containers file loc resolution_acc dir r =
     let file_options = Options.file_options options in
+    let root = get_root options in
     lazy_seq
       [
         lazy
           ( if SSet.mem dir node_modules_containers then
-            lazy_seq
-              ( Files.node_resolver_dirnames file_options
-              |> Core_list.map ~f:(fun dirname ->
-                     lazy
-                       (resolve_relative
-                          ~options
-                          ~reader
-                          loc
-                          ?resolution_acc
-                          dir
-                          (spf "%s%s%s" dirname Filename.dir_sep r))) )
-          else
-            None );
+            lazy_seq([
+              lazy (
+                lazy_seq
+                  ( Files.node_resolver_aliases file_options
+                  |> Core_list.map ~f:(fun dirname ->
+                        let modified_dirname =
+                          dirname
+                          |> Str.split_delim Files.project_root_token
+                          |> String.concat root
+                        in
+                        lazy
+                          (resolve_relative
+                            ~options
+                            ~reader
+                            loc
+                            ?resolution_acc
+                            dir (spf "%s%s%s" modified_dirname Filename.dir_sep r))) )
+              );
+              lazy (
+                lazy_seq
+                  ( Files.node_resolver_dirnames file_options
+                  |> Core_list.map ~f:(fun dirname ->
+                        lazy
+                          (resolve_relative
+                            ~options
+                            ~reader
+                            loc
+                            ?resolution_acc
+                            dir (spf "%s%s%s" dirname Filename.dir_sep r))) )
```

Comment:
I can implement the same `modified_dir` logic here if it's also preferred to support `<PROJECT_ROOT>` for `module.system.node.resolve_dirname`